src/collection-view.js

```diff
@@ -158,6 +158,62 @@ const CollectionView = Backbone.View.extend({
     }
   },
 
+  _addChildModels(models) {
+    let buffer = document.createDocumentFragment();
+    let triggerOnChildren = [];
+
+    this._startUpdating();
+    const addedViews = _.reduce(models, (adding, model, index) =>{
+      let view = this.children.findByModel(model);
+
+      if (!view) {
+        this._destroyEmptyView();
+
+        view = this._addChild(model, index);
+        adding.push(view);
+        triggerOnChildren.push(view);
+        buffer.appendChild(view.el);
+      } else if (buffer.children.length) {
+        if (this.sort) {
+          view._index = index;
+        }
+
+        _.each(triggerOnChildren, child => {
```

Comment:
only if async (which can be said about most function calls)
or if an event call updates the collection in the middle of this function call.
I, honestly, don't like how most of this looks. It's not very DRY because of the `view.$el.before(buffer);` making it difficult to DRY. I am still kicking around how to make it nicer.